### DISCUSSION TITLE
Separate chainstore and statestore

### DISCUSSION THREAD

**Discussion by strahe** [filecoin-project/lotus#7906] [Ideas] [answered]
I tried using a different store for chain store and state store but it doesn't work.

Are there plans to completely separate them?

**Comment by LexLuthr** [comment]
@strahe Can you please clarify a bit more as to what you mean by chainstore and statestore? Maybe provide the output from the terminal?

**Comment by strahe** [comment]
![图片](https://user-images.githubusercontent.com/7932644/150050801-f99b68f7-3186-43eb-b82b-277defa3cd31.png)

At this time, both the state and chain blockstores are backed by the universal store.

I tried building them into 2 different badger instances, couldn't work. 

![图片](https://user-images.githubusercontent.com/7932644/150051327-49f262e4-3dbc-4ec8-804a-167ad7b0606a.png)

Actually what I'm going to describe is this [Todo](https://github.com/filecoin-project/lotus/blob/556e97af17bad9d41f48ac9bbfaac56e94d9aae7/chain/store/snapshot.go#L47), is there a plan to fix it, maybe I can try to submit a pr.

**Reply by LexLuthr** [reply]
@strahe Yes, a PR would be the correct way to go here. You can submit a PR for the TODO in question if you are able to separate the 2. But if you need lotus team to fix the TODO then we can convert this discussion back to an issue with more your suggestion of code changes.

**Comment by strahe** [comment]
@LexLuthr I tried to fix this and everything is fine, but the old snapshots will not be compatible, there are major changes,(maybe there are better ideas), it is better for the Lotus team to fix this. 

https://github.com/strahe/lotus/commit/201e83495f1dc79d0aaa5fc4d38da5e750d846cc

**Reply by Wondertan** [reply]
@strahe, are you aware of splitstore?

**Reply by strahe** [reply]
@Wondertan yes, but they're not the same thing

**Comment by LexLuthr** [comment]
@Reiers Hey, can you get this back into the issues and get it triaged and set a priority to set clear expectations if and when we are going to fix it?

**Comment by LexLuthr** [comment, marked as answer]
@strahe  #8176  has been created to track this. You can add any new inputs there.
Let me know if you would like to keep this discussion open or if we should close it out.

**Reply by strahe** [reply]
Free to close it, thanks.

**Comment by LexLuthr** [comment]
I am archiving this discussion as it has been moved back to an issue.
